pyface/action/listening_action.py

```diff
@@ -55,11 +56,15 @@ def destroy(self):
         """
 
         if self.object:
-            self.object.on_trait_change(
-                self._enabled_update, self.enabled_name, remove=True
+            self.object.observe(
+                self._enabled_update,
+                trait(self.enabled_name, optional=True),
+                remove=True
             )
-            self.object.on_trait_change(
-                self._visible_update, self.visible_name, remove=True
+            self.object.observe(
+                self._visible_update,
+                trait(self.visible_name, optional=True),
```

Comment:
`trait` here assumes the first argument is a trait name (see [doc](https://docs.enthought.com/traits/traits_api_reference/traits.observation.html#traits.observation.expression.trait)), so an extended name like `"child.is_visible"` is going to fail because `"child.is_visible"` as a string is not an attribute on `object`. The `optional=True` may mask that lost behaviour?